1-Draft/RFCNNNN-Improve-Background-Operator.md

```diff
@@ -0,0 +1,208 @@
+with the job data from any stream appearing as it is output in the overlay,
+prefixed with "[JobName]:" (e.g. [Job1]: ). This overlay would be slightly
+larger than Write-Progress so that users could see more output from the job in
+the overlay, and when all jobs sharing output complete the overlay would
+disappear after a delay (5 seconds). Progress messages from the jobs would
+appear above the output.
+
+### Launching a background job quietly and referencing the job with the `$!` variable
+
+```powershell
+# These commands:
+$null = Get-Process &!
+$!
+# Would return the same output as these commands:
+$null = Start-Job {Get-Process}
+$!
+```
+
+```output
+Id     Name            PSJobTypeName   State         HasMoreData     Location             Command
+--     ----            -------------   -----         -----------     --------             -------
+1      Job1            BackgroundJob   Running       True            localhost            Microsoft.PowerShell.Man…
+```
+
+The `$!` variable in those two commands returns the job that was created with
+the command before it.
+
+The `&!` operator permits launching a job using an operator without any of the
+messages from the various streams appearing by default in the console.
+
+### Fanning out to multiple background jobs and referencing the jobs with the `$!` variable
+
+```powershell
+$null = cmd1 & cmd2 & cmd3 &
+$!
+```
+
+```output
+Id     Name            PSJobTypeName   State         HasMoreData     Location             Command
+--     ----            -------------   -----         -----------     --------             -------
+1      Job1            BackgroundJob   Running       True            localhost            Microsoft.PowerShell.Man...
+2      Job1            BackgroundJob   Running       True            localhost            Microsoft.PowerShell.Man...
+3      Job1            BackgroundJob   Running       True            localhost            Microsoft.PowerShell.Man...
+```
+
+The `$!` variable in those two commands returns the jobs that were created with
+the command before it.
+
+### Launching a legacy command with the stop-parsing sigil and either the `&` or `&!` background operator
+
+This corrects a limitation with the background operator(s), which currently
+treat `&` (and `&!`) as an argument instead when used after the stop-parsing
+sigil.
+
+```powershell
+./plink.exe" --% $Hostname -l $Username -pw $Password $Command &
+```
+
+```output
+Id     Name            PSJobTypeName   State         HasMoreData     Location             Command
+--     ----            -------------   -----         -----------     --------             -------
+1      Job1            BackgroundJob   Running       True            localhost            Microsoft.PowerShell.Man...
+```
+
+### Showing or hiding the data from one or more jobs
+
+This allows job data to be shown or hidden on demand for any jobs.
+
+```powershell
+# Show the data from job ids 1-5
+Show-JobData -Id 1..5
```

Comment:
If you change `-ShowInHost`  to `-ShowProgress` then this should be changed to `Show-JobProgress` or `Show-JobStatus`. This way the behavior is consistent.

BTW, it would be cool down the road to figure out how to integrate tmux and terminal split panes. Then we could do `Show-JobProgress -RightPane` and have it automatically split and run the command there. The overlay could be more graphical or detailed at that point.

Reply:
As mentioned in the other reply, I standardized on `Output` as the suffix (`-ShowOutput` parameter name, `Show-/Hide-JobOutput` cmdlets, and a `ShowOutput` property on jobs). That should do the trick.

On the comments about split panes, that would be very cool. :)